### PR TITLE
feat!: initial generation of library

### PR DESCRIPTION
**Pull request by bcoe**

### CODE REVIEW

**Comment by alexander-fenster** [comment]
nit: I think we don't need to have replacements to `s.replace("src/index.js"` in `synth.py`.

File: .jsdoc.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/*!
```
Comment (JustinBeckwith) [inline .jsdoc.js]: We had updated all the other files to use the `//` style comments for the license header.  Could I trouble you to update the template to be consistent?

File: linkinator.config.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "recurse": true,
+  "skip": [
+    "https://codecov.io/gh/googleapis/",
+    "www.googleapis.com",
```
Comment (JustinBeckwith) [inline linkinator.config.json]: Let's add a follow issue so we don't forget to clear these out

File: samples/package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "nodejs-secret-manager-samples",
+  "private": true,
+  "license": "Apache-2.0",
+  "files": [
+    "*.js"
+  ],
+  "author": "Google Inc.",
```
Comment (JustinBeckwith) [inline samples/package.json]: Google LLC.  One of those things we probably need to update elsewhere too.

File: samples/quickstart.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
+/**
```
Comment (JustinBeckwith) [inline samples/quickstart.js]: Can we use the new `//` style for the license header here?

File: samples/quickstart.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+async function main(
```
Comment (JustinBeckwith) [inline samples/quickstart.js]: Can we get a header for samplegen here so users know the arguments?

File: samples/test/quickstart.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+/**
```
Comment (JustinBeckwith) [inline samples/test/quickstart.js]: header

File: synth.py [hunk 7]
```diff
@@ -21,17 +21,17 @@
...
+versions = ['v1beta1']
+for version in versions:
+  library = gapic.typescript_library(
+    'secretmanager',
+    generator_args={
+      "grpc-service-config": "google/cloud/secrets/v1beta1/secretmanager_grpc_service_config.json"
+    },
+    proto_path='/google/cloud/secrets/v1beta1',
+    version=version)
+s.copy(library, excludes=['package.json', 'README.md', 'system-test/fixtures/sample/src/index.js', 'system-test/fixtures/sample/src/index.ts'])
```
Comment (JustinBeckwith) [inline synth.py]: The new generator is creating system tests?  What's up here?

File: .jsdoc.js (hunk 1 above)
Comment (bcoe) [reply .jsdoc.js]: https://github.com/googleapis/synthtool/pull/351

File: linkinator.config.json (hunk 2 above)
Comment (bcoe) [reply linkinator.config.json]: https://github.com/googleapis/nodejs-secret-manager/issues/5

**Pull request merged**